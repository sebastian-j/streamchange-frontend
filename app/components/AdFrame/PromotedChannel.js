import React from 'react';
import PropTypes from 'prop-types';
import styled, { keyframes } from 'styled-components';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const Slide = keyframes`
  0% {
    top: 33vh;
  }
  50% {
    top: 10vh;
  }
  100% {
    top: 33vh;
  }
`;

const AdTitle = styled.span`
  color: #7b7b7b;
  font-family: Arial, sans-serif;
  margin-bottom: 5px;
`;

const PromotedContentWrapper = styled.div`
  animation: ${Slide} 240s linear infinite;
  position: relative;
`;

const PromotedContent = styled.div`
  border: 1px solid #cecece;
  color: ${(props) => props.theme.staticTextColor};
  display: flex;
  flex-direction: row;
  padding: 14px;
`;

const PromotedChannelLogo = styled.img`
  padding-right: 5px;
  width: 7vw;
`;

const PromotedChannelTitle = styled.a`
  color: ${(props) => props.theme.staticTextColor};
  font-weight: bold;
  font-size: 1.3rem;
  margin-bottom: 4px;
  text-decoration: none;
`;

const PromotedChannel = (props) => (
  <PromotedContentWrapper>
    <AdTitle>
      <FormattedMessage {...messages.title} />
    </AdTitle>
    <PromotedContent>
      <a href={props.channelUrl} target="_blank">
        <PromotedChannelLogo alt={props.title} src={props.imageUrl} />
      </a>
      <div>
        <PromotedChannelTitle href={props.channelUrl} target="_blank">
          {props.title}
        </PromotedChannelTitle>
        <div>{props.description}</div>
      </div>
    </PromotedContent>
  </PromotedContentWrapper>
);

PromotedChannel.propTypes = {
  channelUrl: PropTypes.string.isRequired,
  description: PropTypes.string,
  imageUrl: PropTypes.string.isRequired,
  title: PropTypes.string,
};

export default PromotedChannel;
