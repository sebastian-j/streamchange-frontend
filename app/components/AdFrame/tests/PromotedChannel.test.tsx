import React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import PromotedChannel from '../PromotedChannel';

const shallowRenderer = createRenderer();

describe('<PromotedChannel />', () => {
  it('should render and match the snapshot', () => {
    shallowRenderer.render(
      <PromotedChannel
        channelUrl="https://www.youtube.com/channel/UC3GumCi7taJQ0wWbKK-hR2w"
        imageUrl="https://yt3.ggpht.com/ytc/AAUvwnjMKD2nkjne8W3-jn2yCf-lOdD2P1wTJ0Q_G9X72w=s88-c-k-c0x00ffffff-no-rj"
      />,
    );
    const renderedOutput = shallowRenderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
