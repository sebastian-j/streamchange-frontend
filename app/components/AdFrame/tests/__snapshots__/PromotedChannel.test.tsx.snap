// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PromotedChannel /> should render and match the snapshot 1`] = `
<styled.div>
  <styled.span>
    <Memo(MemoizedFormattedMessage)
      defaultMessage="Advertisement"
      id="streamchange.components.AdFrame.title"
    />
  </styled.span>
  <styled.div>
    <a
      href="https://www.youtube.com/channel/UC3GumCi7taJQ0wWbKK-hR2w"
      target="_blank"
    >
      <styled.img
        src="https://yt3.ggpht.com/ytc/AAUvwnjMKD2nkjne8W3-jn2yCf-lOdD2P1wTJ0Q_G9X72w=s88-c-k-c0x00ffffff-no-rj"
      />
    </a>
    <div>
      <styled.a
        href="https://www.youtube.com/channel/UC3GumCi7taJQ0wWbKK-hR2w"
        target="_blank"
      />
      <div />
    </div>
  </styled.div>
</styled.div>
`;
